### PR TITLE
Ports baystation runtime fixes for synth instruments

### PR DESCRIPTION
**Pull request by quardbreak**

### CODE REVIEW

File: code/modules/synthesized_instruments/real_instruments.dm [hunk 1]
```diff
@@ -283,7 +283,8 @@
 
 
 /obj/item/synthesized_instrument/ui_interact(mob/user, ui_key = "instrument", var/datum/nanoui/ui = null, var/force_open = 0)
-	real_instrument.ui_call(user,ui_key,ui,force_open)
+	if (real_instrument)
+		real_instrument.ui_call(user,ui_key,ui,force_open)
```
Comment (afterthought2) [inline code/modules/synthesized_instruments/real_instruments.dm:287]: This proc should not be called on deleted instruments: if it is, that's an explicit call something is making illegally. Are there other ways `real_instrument` can be null?

File: code/modules/synthesized_instruments/song_editor.dm [hunk 2]
```diff
@@ -17,15 +17,14 @@
 
 
 /datum/nano_module/song_editor/proc/current_page()
-	return src.song.current_line > 0 ? Ceiling(src.song.current_line / GLOB.musical_config.song_editor_lines_per_page) : src.page
+	return src.song.current_line > 0 ? Ceiling(src.song.current_line / GLOB.musical_config.song_editor_lines_per_page) : min(src.page, pages())
```
Comment (afterthought2) [inline code/modules/synthesized_instruments/song_editor.dm:20]: Seems superficial: if `pages` is decreased, `page` should be modified accordingly instead.

File: code/modules/synthesized_instruments/song_editor.dm [hunk 3]
```diff
@@ -17,15 +17,14 @@
...
 
 
 /datum/nano_module/song_editor/proc/page_bounds(page_num)
 	return list(
-		max(1 + GLOB.musical_config.song_editor_lines_per_page * (page_num-1), 1),
+		max(min(1 + GLOB.musical_config.song_editor_lines_per_page * (page_num-1), src.song.lines.len), 1),
```
Comment (afterthought2) [inline code/modules/synthesized_instruments/song_editor.dm:25]: `Clamp`, though to be honest I don't see how this is an improvement.

**Comment by MistakeNot4892** [comment]
Closing as eckff is AWOL, feel free to reopen.

**Pull request closed without merging**